app/services/outage-notifications.ts

```diff
@@ -0,0 +1,124 @@
+import { Service } from 'services/service';
+import { NotificationsService, ENotificationType } from 'services/notifications';
+import { Inject } from 'util/injector';
+import { JsonrpcService, IJsonRpcRequest } from 'services/jsonrpc';
+import electron from 'electron';
+
+interface IOutageNotification {
+  /**
+   * Uniquely identifies this message
+   */
+  id: string;
+
+  /**
+   * The body of the message
+   */
+  message: string;
+
+  /**
+   * If set, clicking the notification will
+   * open a browser and navigate to this URL.
+   */
+  url?: string;
+
+  /**
+   * This message should be ignored if disabled is true
+   */
+  disabled: boolean;
+}
+
+// Configuration
+const S3_BUCKET = 'streamlabs-obs';
+const S3_KEY = 'outage-notification.json';
+const POLLING_INTERVAL = 5 * 60 * 1000;
+
+export class OutageNotificationsService extends Service {
+  @Inject() notificationsService: NotificationsService;
+  @Inject() jsonrpcService: JsonrpcService;
+
+  currentMessageId: string = null;
+  currentNotificationId: number = null;
+
+  init() {
+    this.checkForNotification();
+    setInterval(() => this.checkForNotification(), POLLING_INTERVAL);
+  }
+
+  pushNotification(message: string, url?: string) {
+    let action: IJsonRpcRequest;
+
+    action = this.jsonrpcService.createRequest(
+      Service.getResourceId(this.notificationsService),
+      'showNotifications'
+    );
+
+    if (url) {
+      action = this.jsonrpcService.createRequest(
+        Service.getResourceId(this),
+        'openBrowserWindow',
+        url
+      );
+    }
+
+    return this.notificationsService.push({
+      message,
+      type: ENotificationType.WARNING,
+      lifeTime: -1,
+      action
+    });
+  }
+
+  openBrowserWindow(url: string) {
+    electron.remote.shell.openExternal(url);
+  }
+
+  async checkForNotification() {
+    const msg = await this.fetchMessageJson();
+
+    // There are no urgent messages to display to the user
+    if (!msg || msg.disabled) {
+      this.clearNotification();
+      return;
+    }
+
+    // The current message is still in effect
+    if (this.currentMessageId === msg.id) return;
+
+    // There is a new message to display to the user
+    const notification = this.pushNotification(msg.message, msg.url);
+    this.currentMessageId = msg.id;
+    this.currentNotificationId = notification.id;
+  }
```

Comment:
looks like that should be a private method

Reply:
Yeah actually pretty much all of these methods should be.  This service should be treated as a black box that the rest of the app doesn't need to call into.